### PR TITLE
 added command line flag to enable fork-choice late block reorgs

### PR DESCRIPTION
**Pull request by rolfyone**
Currently not enabled, just the flag, and defaulting to false, and a test that shows it can be toggled on.

 partially addresses #6595

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java [hunk 1]
```diff
@@ -113,6 +114,7 @@ public ForkChoice(
       final MergeTransitionBlockValidator transitionBlockValidator,
       final boolean forkChoiceUpdateHeadOnBlockImportEnabled,
       final boolean forkChoiceProposerBoostUniquenessEnabled,
+      final boolean forkChoiceLateBlockReorgEnabled,
       final MetricsSystem metricsSystem) {
```
Comment (lucassaldanha) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java:118]: I understand that this is out of scope for this PR (and should be done in a different PR if ever). But we could have a `ForkChoiceParams` object instead of adding flags to the constructor. Imagine how this is going to look when we have 10 different toggles for the ForkChoice? :)

**Review by lucassaldanha — APPROVED** [review]
LGTM

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java (hunk 1 above)
Comment (rolfyone) [reply ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java:118]: at that point it'll probably annoy me enough to refactor :)
